Sources/Transport/Streams/FoundationStream.swift

```diff
@@ -100,14 +100,15 @@
     }
 
     extension FoundationStream {
-        public func upgradeSSL() {
-            [input, output].forEach { stream in stream.upgradeSSL() }
+        public func upgradeSSL(config: TLS.Config?) {
+            [input, output].forEach { stream in stream.upgradeSSL(config: config) }
         }
     }
 
     extension Foundation.Stream {
         @discardableResult
-        func upgradeSSL() -> Bool {
+        func upgradeSSL(config: TLS.Config?) -> Bool {
+            //TODO: apply TLS.Config's properties to the stream
```

Comment:
I'm ok merging this, but would you mind filing an issue so we don't forget it, I can look into this one since I have some xp w/ NSStreams